robolectric/src/test/java/org/robolectric/shadows/ShadowUIModeManagerTest.java

```diff
@@ -307,4 +310,39 @@ private void setPermissions(String... permissions) {
     pi.requestedPermissions = permissions;
     ((ShadowPackageManager) Shadow.extract(context.getPackageManager())).installPackage(pi);
   }
+
+  @Test
+  @Config(minSdk = O)
+  public void uiModeManager_activityContextEnabled_differentInstancesRetrieveSameValues() {
+    String originalProperty = System.getProperty("robolectric.createActivityContexts", "");
+    System.setProperty("robolectric.createActivityContexts", "true");
+    Activity activity = null;
+    try {
+      UiModeManager applicationUiModeManager =
+          (UiModeManager)
+              ApplicationProvider.getApplicationContext().getSystemService(Context.UI_MODE_SERVICE);
+      activity = Robolectric.setupActivity(Activity.class);
+      UiModeManager activityUiModeManager =
+          (UiModeManager) activity.getSystemService(Context.UI_MODE_SERVICE);
+
+      assertThat(applicationUiModeManager).isNotSameInstanceAs(activityUiModeManager);
+
+      int applicationNightMode = applicationUiModeManager.getNightMode();
+      int activityNightMode = activityUiModeManager.getNightMode();
+
+      assertThat(activityNightMode).isEqualTo(applicationNightMode);
+
+      int testMode = UiModeManager.MODE_NIGHT_YES;
```

Comment:
Maybe there should be a check to ensure that the current mode is not already `UiModeManager.MODE_NIGHT_YES`?